qiskit_experiments/framework/json.py

```diff
@@ -0,0 +1,516 @@
+    """Deserialize binary encoded data.
+
+    Args:
+        data: Data to be serialized.
+
+    Returns:
+        String representation.
+    """
+    try:
+        encoded = value["encoded"]
+        compressed = value["compressed"]
+        decoded = base64.standard_b64decode(encoded)
+        if compressed:
+            decoded = zlib.decompress(decoded)
+        return decoded
+    except Exception as ex:  # pylint: disable=broad-except
+        warning_msg = "Could not deserialize binary encoded data."
+        traceback_msg = traceback.format_exception(type(ex), ex, ex.__traceback__)
+        _show_warning(warning_msg, traceback_msg=traceback_msg)
+        return value
+
+
+def _serialize_and_encode(
+    data: Any, serializer: Callable, compress: bool = True, **kwargs: Any
+) -> str:
+    """Serialize the input data and return the encoded string.
+
+    Args:
+        data: Data to be serialized.
+        serializer: Function used to serialize data.
+        compress: Whether to compress the serialized data.
+        kwargs: Keyword arguments to pass to the serializer.
+
+    Returns:
+        String representation.
+    """
+    buff = io.BytesIO()
+    serializer(buff, data, **kwargs)
+    buff.seek(0)
+    serialized_data = buff.read()
+    buff.close()
+    value = _serialize_bytes(serialized_data, compress=compress)
+    return value
+
+
+def _decode_and_deserialize(value: Dict, deserializer: Callable, name: Optional[str] = None) -> Any:
+    """Decode and deserialize input data.
+
+    Args:
+        value: The binary encoded serialized data value.
+        deserializer: Function used to deserialize data.
+        name: Object type name for warning message if deserialization fails.
+
+    Returns:
+        Deserialized data.
+    """
+    try:
+        buff = io.BytesIO()
+        buff.write(value)
+        buff.seek(0)
+        orig = deserializer(buff)
+        buff.close()
+        return orig
+    except Exception as ex:  # pylint: disable=broad-except
+        warning_msg = f"Could not deserialize <{name}> data."
+        traceback_msg = traceback.format_exception(type(ex), ex, ex.__traceback__)
+        _show_warning(warning_msg, traceback_msg=traceback_msg)
+        return value
+
+
+def _serialize_safe_float(obj: any):
+    """Recursively serialize basic types safely handing inf and NaN"""
+    if isinstance(obj, float):
+        if math.isfinite(obj):
+            return obj
+        else:
+            value = obj
+            if math.isnan(obj):
+                value = "NaN"
+            elif obj == math.inf:
+                value = "Infinity"
+            elif obj == -math.inf:
+                value = "-Infinity"
+            return {"__type__": "safe_float", "__value__": value}
+    elif isinstance(obj, (list, tuple)):
+        return [_serialize_safe_float(i) for i in obj]
+    elif isinstance(obj, dict):
+        return {key: _serialize_safe_float(val) for key, val in obj.items()}
```

Comment:
This will not deserialize to the same object. The default python JSON serializer already converts some keys to strings implicitly (like ints). But for general hashable types I think this should raise an exception instead of converting that type to a string which is not reversible.

Reply:
I see. I was assuming somewhat of old qobj schema.